src/components/Widget/transactions.ts

```diff
@@ -1,17 +1,76 @@
 import { TransactionReceipt } from '@ethersproject/abstract-provider'
-import { TransactionEventHandlers } from '@uniswap/widgets'
-import { useMemo } from 'react'
+import { TradeType, Transaction, TransactionEventHandlers, TransactionInfo, TransactionType } from '@uniswap/widgets'
+import { useWeb3React } from '@web3-react/core'
+import { V3_SWAP_DEFAULT_SLIPPAGE } from 'hooks/useAutoSlippageTolerance'
+import { useCallback, useMemo } from 'react'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import {
+  ExactInputSwapTransactionInfo,
+  ExactOutputSwapTransactionInfo,
+  WrapTransactionInfo,
+} from 'state/transactions/types'
+import { useUserSlippageTolerance } from 'state/user/hooks'
+import { currencyId } from 'utils/currencyId'
 
 /** Integrates the Widget's transactions, showing the widget's transactions in the app. */
 export function useSyncWidgetTransactions() {
-  // TODO(jfrankfurt): Integrate widget transactions with app transaction tracking.
+  const { chainId } = useWeb3React()
+  const addTransaction = useTransactionAdder()
+  const [appSlippage] = useUserSlippageTolerance()
+  const allowedSlippage = appSlippage === 'auto' ? V3_SWAP_DEFAULT_SLIPPAGE : appSlippage
+
+  const onTxSubmit = useCallback(
+    (_hash: string, transaction: Transaction<TransactionInfo>) => {
+      const { type, response } = transaction.info
+
+      if (!type || !response) {
+        return
+      }
+      if (type === TransactionType.WRAP || type === TransactionType.UNWRAP) {
+        const { amount } = transaction.info
+
+        addTransaction(response, {
+          type: TransactionType.WRAP,
+          unwrapped: false,
```

Comment:
This should be dependent on `TransactionType`, maybe:

    unwrapped: type === TransactionType.UNWRAP